src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeNameParserHelpers.cs

```diff
@@ -135,6 +163,50 @@ static int GetUnescapedOffset(ReadOnlySpan<char> fullName, int startIndex)
             }
         }
 
+        internal static string Unescape(string input)
+        {
+            int indexOfEscapeChar = input.IndexOf(EscapeCharacter);
+            if (indexOfEscapeChar < 0)
+            {
+                // Nothing to escape, just return the original value.
+                return input;
+            }
+
+            ValueStringBuilder builder = new(stackalloc char[128]);
```

Comment:
stackalloc is relatively expensive when it is not reachable. It should be in the helper method too.

Reply:
Done.